client/pages/NewTrip/NewTrip.jsx

```diff
@@ -100,30 +107,45 @@ class NewTrip extends Component {
   }
 
   getPointName = (startPoint) => {
-    const { google } = this.props;
     const name = new Promise((resolve, reject) => {
-      if (!google) {
+      if (!window.google) {
         reject(new Error('API is undefined'));
       }
-      const geocoder = new google.maps.Geocoder();
+      const geocoder = new window.google.maps.Geocoder();
       geocoder.geocode({ location: startPoint }, res => resolve(
         this.getParsetNameFromGeocode(res[0])
       ));
     });
     return name;
   }
 
-  addMarkers = (latLng) => {
+  addMarkers = (latLng, pointFlag) => {
     const { markers } = this.state;
-    if (markers.length > 1) {
-      this.setState(prevState => prevState.markers.pop());
+    let flag = markers.length ? 'B' : 'A';
+    if (pointFlag) {
+      flag = pointFlag;
+      this.setPoints(latLng, pointFlag);
+    }
+    if (pointFlag === 'A') {
+      this.getPointName(latLng)
+        .then(res => this.setState(prevState => prevState.markers.splice(0, 1, {
+          name: res,
+          title: res,
+          position: latLng,
+          point: flag
+        })));
+    } else {
```

Comment:
What if you what to add more than two points?

Reply:
Right now we have functional for two points, and we don't need to add more.